src/main/java/bloop/Parser.java

```diff
@@ -0,0 +1,64 @@
+package bloop;
+
+import java.io.IOException;
+
+/**
+ * Parses the input and performs relevent operations.
+ */
+public class Parser {
+
+    private TaskList tasks;
+
+    /**
+     * Constructor for Parser object.
+     *
+     * @param tasks Tasklist object.
+     */
+    public Parser(TaskList tasks) {
+        this.tasks = tasks;
+    }
+
+    /**
+     * Parses the input by the user.
+     *
+     * @param text Input by the user.
+     * @throws BloopException If the command is not recognised or incomplete.
+     * @throws IOException If there is a problem writing to file.
+     */
+    public void parse(String text) throws BloopException, IOException {
```

Comment:
According to the code-quality guide, you should do something if a function is more than 30 LOC.